Oracle 12c data loading support

mirasrael opened the pull request:
Support loading data in pluggable database for Oracle 12c

cataphract commented:
AFAIK, what comes after `/` is the service name; it's not specific to PDB's. The naming is therefore off.

Since SID is on the way out, it could make more sense to _replace_ `$ORASID`.

But most of all, I'm not satisfied with one method being used in one place and another in other places. It will just generate confusion:

```
$ git grep ORASID
config/DataSource.groovy.php:    url             = 'jdbc:oracle:thin:@<?= $_ENV['ORAHOST'] ?>:<?= $_ENV['ORAPORT'] ?>:<?= $_ENV['ORASID'] ?>'
data/oracle/Common.properties.php:url=jdbc:oracle:thin:@<?= $_ENV['ORAHOST'] ?>:<?= $_ENV['ORAPORT'] ?>:<?= $_ENV['ORASID'], "\n" ?>"
lib/inc/oracle/SqlProducer.groovy:        def url = "jdbc:oracle:thin:@${getenv 'ORAHOST'}:${getenv 'ORAPORT'}:${getenv "ORASID"}"
samples/common/dump_audit.groovy:    jdbcUrl = "jdbc:oracle:thin:@${e.ORAHOST}:${e.ORAPORT}:${e.ORASID}"
samples/oracle/DatabaseConnection.groovy:  String jdbcUrl = "jdbc:oracle:thin:@${System.getenv('ORAHOST')}:${System.getenv('ORAPORT')}:${System.getenv('ORASID')}"
samples/oracle/gen_kettle_properties.php:TM_CZ_DB_NAME=<?= $_ENV['ORASID'], "\n" ?>
samples/oracle/gen_kettle_properties.php:TM_LZ_DB_NAME=<?= $_ENV['ORASID'] , "\n" ?>
samples/oracle/gen_kettle_properties.php:TM_WZ_DB_NAME=<?= $_ENV['ORASID'] , "\n" ?>
samples/oracle/gen_kettle_properties.php:DEAPP_DB_NAME=<?= $_ENV['ORASID'] , "\n" ?>
samples/oracle/gen_kettle_properties.php:BIOMART_DB_NAME=<?= $_ENV['ORASID'] , "\n" ?>
samples/oracle/gen_kettle_properties.php:BIOMART_STAGE_DB_NAME=<?= $_ENV['ORASID'] , "\n" ?>
samples/oracle/gen_kettle_properties.php:I2B2DEMODATA_DB_NAME=<?= $_ENV['ORASID'] , "\n" ?>
solr/data-config/oracle/data-config_browse.xml.php:            url="jdbc:oracle:thin:@<?= $_ENV['ORAHOST'] ?>:<?= $_ENV['ORAPORT'] ?>:<?= $_ENV['ORASID'] ?>"
solr/data-config/oracle/data-config_rwg.xml.php:              url="jdbc:oracle:thin:@<?= $_ENV['ORAHOST'] ?>:<?= $_ENV['ORAPORT'] ?>:<?= $_ENV['ORASID'] ?>"
solr/data-config/oracle/data-config_sample.xml.php:            url="jdbc:oracle:thin:@<?= $_ENV['ORAHOST'] ?>:<?= $_ENV['ORAPORT'] ?>:<?= $_ENV['ORASID'] ?>"
vars.sample:ORASID=orcl
vars.sample: ORASID ORAUSER ORAPASSWORD ORACLE_MANAGE_TABLESPACES \
```

mirasrael commented:
I renamed option to ORASVC to better reflect it meaning. I also changed some php templates (related to SOLR), but I'm not familar with kettle, so I don't know how to save change gen_kettle_properties template. Also it is not clear for me where Common.properties and dump_audit.groovy used to be able to fix and test them.

cataphract commented:
Merged in 0cfbf685fc9b4a3d0027adb2a8f68874f3c2fdff.
